apollo-federation/src/merger/merger.rs

```diff
@@ -1028,8 +1029,124 @@ impl Merger {
         Ok(!source_as_entity.is_empty())
     }
 
-    pub(crate) fn merge_interface(&mut self, _itf: InterfaceTypeDefinitionPosition) {
-        todo!("Implement merge_interface")
+    fn validate_interface_keys(
+        &mut self,
+        sources: &Sources<Subgraph<Validated>>,
+        dest: &InterfaceTypeDefinitionPosition,
+    ) -> Result<bool, FederationError> {
+        let supergraph_implementations = self.merged.possible_runtime_types(dest.clone().into())?;
+
+        let mut has_key = false;
+        for (idx, source) in sources.iter() {
+            if source.is_none()
+                || self.subgraphs[*idx].is_interface_object_type(&dest.clone().into())
+            {
+                continue;
+            }
+
+            let subgraph = source.as_ref().unwrap();
+            let source_metadata = self.subgraphs[*idx].metadata();
+            let interface_pos: TypeDefinitionPosition = dest.clone().into();
+            let key_directive_name = source_metadata
+                .federation_spec_definition()
+                .key_directive_definition(&self.merged)?
+                .name
+                .clone();
+            let keys = interface_pos.get_applied_directives(subgraph.schema(), &key_directive_name);
+
+            has_key = has_key || !keys.is_empty();
+            let resolvable_key = keys.iter().find(|key| !key.arguments.is_empty());
+            if resolvable_key.is_none() {
+                continue;
+            }
+
+            let implementations_in_subgraph = subgraph
+                .schema()
+                .possible_runtime_types(dest.clone().into())?;
+            if implementations_in_subgraph.len() > supergraph_implementations.len() {
```

Comment:
this check is flipped and should be checking if it is less than in the supergraph? (https://github.com/apollographql/federation/blob/main/composition-js/src/merging/merge.ts#L2386)